packages/ui/src/components/Toast/Toast.tsx

```diff
@@ -28,74 +28,122 @@ export const toastStyles = cva(
   children,
   className,
-}: ToastProps = {}) => {
-  sonnerToast.custom((t) => (
+}: ToastProps) => {
+  sonnerToast.custom(t => (
     <div className={toastStyles({ colorScheme, className })}>
-      <div>{children}</div>
-      <div className="h-6 border border-outline-low-em" />
-      {action && <div>action</div>}
-      <Button variant="ghost" onClick={() => sonnerToast.dismiss(t)}>
-        <Icon name="cross" />
-      </Button>
+      <div className="flex items-center space-x-4 shrink-0">
+        {children}
+        {actionText && (
+          <Button
+            colorScheme="main"
+            variant="tertiary"
+            size="sm"
+            onClick={actionOnClick}
+          >
+            {actionText}
+          </Button>
+        )}
+        <IconButton
+          variant="ghost"
+          name="cross"
+          size="xs"
+          onClick={() => sonnerToast.dismiss(t)}
+        />
+      </div>
     </div>
   ));
 };
 
-export const errorToast = ({ action, children, className }: ToastProps = {}) =>
+export const errorToast = ({
+  actionOnClick,
+  actionText,
+  children,
+  className,
+  iconName = "info-fill",
+}: ToastProps) =>
   toast({
     children: (
-      <div className="flex items-center space-x-4">
-        <Icon name="info-fill" className=" text-text-danger-main" />
-        <div>{children}</div>
-      </div>
+      <IconWithChildren iconName={iconName}>{children}</IconWithChildren>
     ),
-    action,
+    actionOnClick,
+    actionText,
     colorScheme: "error",
     className,
   });
 
 export const successToast = ({
-  action,
+  actionOnClick,
+  actionText,
   children,
   className,
-}: ToastProps = {}) =>
+  iconName = "tick-fill",
+}: ToastProps) =>
   toast({
     children: (
-      <div className="flex items-center space-x-4">
-        <Icon name="info-fill" className=" text-text-success-main" />
-        <div>{children}</div>
-      </div>
+      <IconWithChildren iconName={iconName}>{children}</IconWithChildren>
     ),
-    action,
+    actionOnClick,
+    actionText,
     colorScheme: "success",
     className,
   });
 
 export const warningToast = ({
-  action,
+  actionOnClick,
+  actionText,
   children,
   className,
-}: ToastProps = {}) =>
+  iconName = "warning-fill",
+}: ToastProps) =>
   toast({
     children: (
-      <div className="flex items-center space-x-4">
-        <Icon name="info-fill" />
-        <div>{children}</div>
-      </div>
+      <IconWithChildren iconName={iconName}>{children}</IconWithChildren>
     ),
-    action,
+    actionOnClick,
+    actionText,
     colorScheme: "warning",
     className,
   });
 
+export const infoToast = ({
+  actionOnClick,
+  actionText,
+  children,
+  className,
+  iconName = "info-fill",
+}: ToastProps) =>
+  toast({
+    children: (
+      <IconWithChildren iconName={iconName}>{children}</IconWithChildren>
+    ),
+    actionOnClick,
+    actionText,
+    colorScheme: "info",
+    className,
+  });
+
 export { Toaster };
```

Comment:
Should we export the `toast` function too? Or we let the developer install sonner on the app?

Reply:
we can export toast

Reply:
it's already exporting above

`export const toast = ({`